src/components/Products.jsx

```diff
@@ -1,27 +1,27 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useContext } from 'react';
 import { Link } from 'react-router-dom';
 import useFetch from '../hooks/useFetch';
+import { ThemeContext } from "../ThemeContext";
+import ToggleSwitch from './ToggleSwitch';
 
 function Products() {
   const [products, setProducts] = useState([]);
   const { data, loading } = useFetch('products');
 
-  useEffect(() => {
-    if (data) {
-      setProducts(data);
-    }
-  }, [data]);
 
   if (loading) {
     return <div>Loading...</div>;
   }
 
+  
+  const{toggleTheme, theme} = useContext(ThemeContext); 
```

Comment:
This breaks the rules of hooks because `useContext` is used conditionally and will only run if we are _not_ in a loading state. VS Code shows you a red underline underneath `useContext`.